doc/ceph-volume/lvm/zap.rst

```diff
@@ -15,18 +15,51 @@ on the given lv or partition will be removed and all data will be purged.
 
 Zapping a logical volume::
 
-      ceph-volume lvm zap {vg name/lv name}
+    ceph-volume lvm zap {vg name/lv name}
 
 Zapping a partition::
 
-      ceph-volume lvm zap /dev/sdc1
+    ceph-volume lvm zap /dev/sdc1
 
-If you are zapping a raw device or partition and would like any vgs or lvs created
-from that device removed use the ``--destroy`` flag. A common use case is to simply
-deploy OSDs using a whole raw device. If you do so and then wish to reuse that device for
-another OSD you must use the ``--destroy`` flag when zapping so that the vgs and lvs that
-ceph-volume created on the raw device will be removed.
+Removing Devices
+----------------
+When zapping, and looking for full removal of the device (lv, vg, or partition)
+use the ``--destroy`` flag. A common use case is to simply deploy OSDs using
+a whole raw device. If you do so and then wish to reuse that device for another
+OSD you must use the ``--destroy`` flag when zapping so that the vgs and lvs
+that ceph-volume created on the raw device will be removed.
+
+.. note:: Multiple devices can be accepted at once, to zap them all
 
 Zapping a raw device and destroying any vgs or lvs present::
 
-      ceph-volume lvm zap /dev/sdc --destroy
+    ceph-volume lvm zap /dev/sdc --destroy
+
+
+This action can be performed on partitions, and logical volumes as well::
+
+    ceph-volume lvm zap /dev/sdc1 --destroy
+    ceph-volume lvm zap osd-vg/data-lv --destroy
+
+
+Finally, multiple devices can be detected if filtering by OSD ID and/or OSD
+FSID. Either identifier can be used or both can be used at the same time. This
+is useful in situations where multiple devices associated with a specific ID
+need to be purged. When using the FSID, the filtering is stricter, and might
+not match other (possibly invalid) devices associated to an ID.
+
+By ID only::
```

Comment:
The reason we support this is to address exactly what you describe: there have been scenarios where an admin has devices from previous deployments that are invalid and need to be removed. Allowing using `--osd-id` only to match is useful there.

If we enforce asking for `--cluster-fsid` then one would have to go and find what that is exactly for each OSD. 

Regardless of these options, full removal of *any* device is dangerous, and there is potential still to destroy devices by mistake that are part of a running cluster

Reply:
Yes, I understand, what I'm saying is that using OSD ID only seems dangerous on a containerized deployment. We need to enforce the usage of OSD FSID so that we are 100% we are targetting the right OSD.

What do you think?

Reply:
I think that it is reasonable to always use --osd-fsid by containers. But it is still useful to have it for non-containers

Reply:
Alright, sounds good.

Reply:
Your concern is valid for non-container scenarios as well. The `zap` sub-command is a destructive one and is potentially dangerous.

If it matches several devices with the same ID it will remove them all.

Reply:
I think we should remove the `--osd-id` option and only have `--osd-fsid`. Using the OSD ID is just too dangerous, also in a near future, all the clusters will be containerized so this is not something we can afford letting go through.

Reply:
Like I mentioned, anything in zap is *really dangerous*. One can still destroy a system by using zap. I understand your concern, we can revisit this once again when most/all Ceph clusters are done with containers. Until then, we must be able to support non-container deployments, to which this specific change allows to remove more than one set of devices associated with an OSD